lib/GenomeFileUtil/core/GenomeUtils.py

```diff
@@ -369,3 +377,107 @@ def sort_dict(in_struct):
+    tax_id = str(tax_id)
+    genome_dict.setdefault('warnings', [])
+    assignments = genome_dict.get('taxon_assignments')
+    # Check to make sure that tax_id == genome.taxon_assignments.ncbi
+    if assignments and assignments.get(_NCBI_TAX) and assignments[_NCBI_TAX] != tax_id:
+        raise RuntimeError(
+            f"The provided taxon ID ({tax_id}) differs from the "
+            f"taxon ID in the genome's `taxon_assignments` field: {assignments[_NCBI_TAX]}."
+        )
+    genome_dict['taxon_assignments'] = {'ncbi': tax_id}
+    # Fetch the taxon from Relation Engine by taxon ID
+    try:
+        resp_json = re_client.stored_query(
+            'ncbi_fetch_taxon',
+            {'id': str(tax_id), 'ts': now},
+            raise_not_found=True)
+    except RENotFound as err:
+        # Taxon not found; log and raise
+        logging.error(str(err))
+        raise err
+    re_result = resp_json['results'][0]
+    # Refer to the following schema for returned fields in `re_result`:
+    # https://github.com/kbase/relation_engine_spec/blob/develop/schemas/ncbi/ncbi_taxon.yaml
+    gencode = int(re_result['gencode'])
+    # If there is a mismatch on some of these fields from NCBI, save a warning and continue
+    if genome_dict.get('genetic_code') and genome_dict['genetic_code'] != gencode:
+        genome_dict['warnings'].append(
+            f"The genetic code provided by NCBI ({gencode}) "
+            f"does not match the one given by the user ({genome_dict['genetic_code']})"
+        )
+    genome_dict['genetic_code'] = gencode
+    # Fetch the lineage on RE using the taxon ID to fill the "taxonomy" and "domain" fields.
+    lineage_resp = re_client.stored_query(
+        'ncbi_taxon_get_lineage',
+        {'id': str(tax_id), 'ts': now, 'select': ['scientific_name', 'rank']},
+        raise_not_found=True)
+    # The results will be an array of taxon docs with "scientific_name" and "rank" fields
+    lineage = [
+        r['scientific_name'] for r in lineage_resp['results']
+        if r['scientific_name'] != 'root' and r['scientific_name'] != 'cellular organisms'
+    ]
+    # Format and normalize the lineage string
+    taxonomy = '; '.join(lineage).replace('\n', '')
+    # Fetch the domain in the lineage. The `domain` var should be a singleton list.
+    # In NCBI taxonomy, 'domain' is known as 'superkingdom'
+    domain = [r['scientific_name'] for r in lineage_resp['results'] if r['rank'] == 'superkingdom']
+    if genome_dict.get('domain') and genome_dict['domain'] != domain:
+        genome_dict['warnings'].append(
+            f"The domain provided by NCBI ({domain}) "
+            f"does not match the one given by the user ({genome_dict['domain']})"
+        )
+    # Set the domain from NCBI, if possible. Otherwise, fall back to anything
+    # the user supplied, or 'Unknown'.
+    if domain:
+        genome_dict['domain'] = domain[0]
+    elif not genome_dict.get('domain'):
```

Comment:
do we want domain to be missing if its not in the lineage query or in the genome_dict?

Reply:
Maybe?